contrib/pkg/provision/buildami.go

```diff
@@ -0,0 +1,91 @@
+package provision
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/openshift/cluster-operator/contrib/pkg/amibuild"
+	"github.com/openshift/cluster-operator/contrib/pkg/amiid"
+)
+
+func (o *ProvisionClusterOptions) BuildAMI(clients *clientContext) error {
+	var err error
+	if len(o.RPMRepositoryURL) == 0 {
+		o.RPMRepositoryURL, err = o.determineRPMRepositoryURL(o.RPMRepositorySource)
+		if err != nil {
+			return fmt.Errorf("cannot retrieve RPM repository URL from %s: %v", o.RPMRepositorySource, err)
+		}
+		o.Logger.WithField("RepositoryURL", o.RPMRepositoryURL).Debug("Obtained RPM repository URL")
+	}
+
+	buildAMIOptions := amibuild.BuildAMIOptions{
+		AccountSecretName:      o.AccountSecretName,
+		SSHSecretName:          o.SSHSecretName,
+		AnsibleImage:           o.AnsibleImage,
+		AnsibleImagePullPolicy: corev1.PullIfNotPresent,
+		RPMRepositoryURL:       o.RPMRepositoryURL,
+		RPMRepositorySource:    o.RPMRepositorySource,
```

Comment:
I'm almost positive I'm missing something here but are these two RPM fields used anywhere? I can't see other than where they're validated.

Reply:
The RepositoryURL is used by the playbook that builds the AMI here: https://github.com/openshift/cluster-operator/blob/master/contrib/pkg/amibuild/generate.go#L63

The RepositorySource is used to determine a RepositoryURL if not specified directly. This command the code from the other utility commands like buildami and waitforcluster.